tests/Doctrine/Tests/DBAL/Functional/ModifyLimitQueryTest.php

```diff
@@ -195,4 +199,33 @@ private function assertLimitResult(
             self::assertCount(count($expectedResults), $data);
         }
     }
+
+    public function testLimitWhenOrderByWithSubqueryWithOrderBy(): void
+    {
+        $platform = $this->connection->getDatabasePlatform();
+        if ($platform instanceof DB2Platform) {
+            self::markTestSkipped('Generated SQL is invalid');
+        }
+
+        $platformName = get_class($platform);
+        if (
+            $platformName === SQLServer2008Platform::class
+            || $platformName === SQLServer2005Platform::class
```

Comment:
The `get_class()` could be avoided if coded as `if ($platform instanceof SQLServerPlatform && ! $platform instanceof SQLServer2012Platform)`.